### PR TITLE
Add API Key support in SDKs

### PR DESCRIPTION
**Pull request by hdkshingala**
### Summary by CodeRabbit

- New Feature: Introduced the ability to configure the Aperture SDK using an API key, enhancing security and flexibility. This change affects all SDKs (Go, Python, JavaScript, and Java).
- New Feature: Added the option to use insecure connections to the Aperture Agent, providing more flexibility in testing and development environments.
- Improvement: Updated the Aperture SDK to use a single `APERTURE_AGENT_ADDRESS` environment variable for the agent's address, simplifying configuration.
- Improvement: Updated the Aperture SDK to use a longer default timeout value for HTTP middleware, enhancing reliability.
- Improvement: Updated the library version to "2.22.0" across all SDKs.
- Documentation: Updated README and example code to reflect the new configuration options.

### CODE REVIEW

**Review by harjotgill — APPROVED** [review]

File: sdks/aperture-java/examples/netty-example/src/main/java/com/fluxninja/example/NettyServer.java [hunk 1]
```diff
@@ -22,6 +22,10 @@ public static void main(String[] args) throws Exception {
         if (agentAddress == null) {
             agentAddress = DEFAULT_AGENT_ADDRESS;
         }
+        String agentAPIKey = System.getenv("APERTURE_AGENT_API_KEY");
+        if (agentAPIKey == null) {
+            agentAPIKey = "";
+        }
```
[the remark this answers is not in the document]
Comment (hdkshingala) [reply sdks/aperture-java/examples/netty-example/src/main/java/com/fluxninja/example/NettyServer.java:28]: API Key is not a required field

File: sdks/aperture-java/examples/standalone-example/src/main/java/com/fluxninja/example/App.java [hunk 2]
```diff
@@ -36,6 +38,10 @@ public static void main(String[] args) {
         if (agentAddress == null) {
             agentAddress = DEFAULT_AGENT_ADDRESS;
         }
+        String agentAPIKey = System.getenv("APERTURE_AGENT_API_KEY");
+        if (agentAPIKey == null) {
+            agentAPIKey = "";
+        }
```
[the remark this answers is not in the document]
Comment (hdkshingala) [reply sdks/aperture-java/examples/standalone-example/src/main/java/com/fluxninja/example/App.java:44]: API Key is not a required field

File: sdks/aperture-java/examples/armeria-example/src/main/java/com/fluxninja/example/ArmeriaServer.java [hunk 3]
```diff
@@ -50,6 +54,10 @@ public static void main(String[] args) {
         if (agentHost == null) {
             agentHost = DEFAULT_AGENT_ADDRESS;
         }
+        String agentAPIKey = System.getenv("APERTURE_AGENT_API_KEY");
+        if (agentAPIKey == null) {
+            agentAPIKey = "";
+        }
```
[the remark this answers is not in the document]
Comment (hdkshingala) [reply sdks/aperture-java/examples/armeria-example/src/main/java/com/fluxninja/example/ArmeriaServer.java:60]: API Key is not a required field

File: sdks/aperture-java/javaagent/agent/src/main/java/com/fluxninja/aperture/instrumentation/Config.java [hunk 4]
```diff
@@ -106,6 +106,7 @@ public static ApertureSDKWrapper newSDKWrapperFromConfig() {
                     builder.setAddress(
                                     config.getProperty(
                                             AGENT_ADDRESS_PROPERTY, AGENT_ADDRESS_DEFAULT_VALUE))
+                            .setAgentAPIKey(config.getProperty("aperture.agent.api.key", ""))
```
[the remark this answers is not in the document]
Comment (hdkshingala) [reply sdks/aperture-java/javaagent/agent/src/main/java/com/fluxninja/aperture/instrumentation/Config.java]: Added constant and exposed it in allProperties.
Updated the README file and I am planning to update the docs separately as there are other changes also required in docs for SaaS agent

**Review by IridiumOxide — APPROVED** [review]

**Pull request merged**